internal/repository/repo.go

```diff
@@ -0,0 +1,117 @@
+package repository
+
+import (
+	model "github.com/cyneruxyz/address-book/gen/proto"
+	"hash/fnv"
+	"strings"
+	"sync"
+)
+
+type AddressBook struct {
+	sync.Mutex
+	Names  map[string][]*model.AddressField
+	Addrs  map[string][]*model.AddressField
+	Phones map[string][]*model.AddressField
+	Hashes map[uint32]*model.AddressField
+}
+
+func NewAddressBook() *AddressBook {
+	return &AddressBook{}
+}
+
+func (ab *AddressBook) CreateAddressField(af *model.AddressField) (ok bool) {
+	hash := getHash(af)
+	if _, ok := ab.Hashes[hash]; ok {
```

Comment:
Here and in the same cases below: what's about the concurrency here? Somebody can try to update `Hashes` during that action